include/boost/beast/websocket/detail/utf8_checker.hpp

```diff
@@ -121,8 +121,8 @@ write(std::uint8_t const* in, std::size_t size)
             {
                 if(    (p[1] & 0xc0) != 0x80
                     || (p[2] & 0xc0) != 0x80
-                    || (p[0] == 0xe0 && (p[1] & 0xe0) == 0x80) // overlong
-                    || (p[0] == 0xed && (p[1] & 0xe0) == 0xa0) // surrogate
+                    || (p[0] == 0xe0 && (p[1] & 0x20) == 0) // overlong
+                    || (p[0] == 0xed && (p[1] & 0x3C) >= 0x3C) // surrogate
```

Comment:
The old surrogate check was wrong...
The check for UTF-16 surrogate pairs should be to verify that the values after 0xED do not fall in the range 0x800...0xFFF. Since, in the second byte of 3-byte sequence, 0x8 is shifted left by 2 bits, the mask becomes 0x3C.